### DISCUSSION TITLE
Can you get more properties at a time with Color Switch command class?

### DISCUSSION THREAD

**Discussion by StreetCube** [zwave-js/zwave-js#5169] [Q&A] [answered]
Right now if I want to get my led bulb's colors it seems the only way to do that is to use the Color Switch command class's get() function. 

In the documentation it seemed like I can only get it one color at a time with the get function since it needs a colorComponent enum value and it doesn't seem to accept objects for a "getBulk (or something like that)". Is this the best/fastest way if I want to poll everything from my Led bulb, or is there a better way that I don't seem to be aware of?

**Comment by kpine** [comment, marked as answer]
The Color Switch CC doesn't support a method to bulk Get method, it's one at a time. If you're using the JS API directly, I think you can use the [`ZWaveNode.refreshCCValues`](https://zwave-js.github.io/node-zwave-js/#/api/node?id=refreshccvalues) method to get all colors in a single call, it just loops over all of the supported colors and does individual Gets.

**Reply by StreetCube** [reply]
So if I understand right, in terms of effectiveness it's gonna do the same thing if I just manually get everything?

**Reply by kpine** [reply]
Correct. There's no Z-Wave command to Get bulk colors (unlike Set, which supports it). You can look at the [spec](https://github.com/zwave-js/specs/blob/master/Z-Wave%20Specification%20AWG%20V1.0.pdf) if you're interested.
